internal/collector/processor/queued_processor_test.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	agenttracepb "github.com/census-instrumentation/opencensus-proto/gen-go/agent/trace/v1"
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+func TestQueueProcessorHappyPath(t *testing.T) {
+	wg := &sync.WaitGroup{}
+	mockProc := &mockConcurrentSpanProcessor{waitGroup: wg}
+	qp := NewQueuedSpanProcessor(mockProc)
+	goFn := func(b *agenttracepb.ExportTraceServiceRequest) {
+		qp.ProcessSpans(b, "test")
+	}
+
+	spans := []*tracepb.Span{{}}
+	wantBatches := 10
+	wg.Add(wantBatches)
```

Comment:
:smile: yes, it terminates (Travis agree). The wait group is passed to the mock and the mock is in charge of calling `wg.Done()`.

Reply:
LOL gotcha, that confused me a bit

Reply:
Later perphaps it would be nice to make mockConcurrentSpanProcessor handle all the synchronization logic otherwise right now we are leaking the abstraction with wg.Add and wg.Wait yet no wg.Done.

Methods perhaps:
* `(mockSpanProcessor) runConcurrently(fn func())` which will handle doing `wg.Add` and then `wg.Done`
* `(mockSpanProcessor) awaitAsyncProcessing()` which then invoke `wg.Wait` 
will ensure that we don't have to expose such internals nor ever have to worry, thinking about termination

Reply:
good suggestion, done.